include/tvm/ir/module.h

```diff
@@ -39,8 +39,52 @@
 #include <utility>
 #include <vector>
 
+namespace tvm {
+namespace tir {
+// Forward declarations from tvm::tir
+class PrimFunc;
+
+/*!
+ * \brief Describes one parameter that should be linked into the generated module.
+ *
+ * When parameters are to be linked in with generated code (i.e. on target_host-compatible
+ * backends), Relay attaches instances of this object to a global TIR function. Code-generators
+ * use the information contained in this node to include the parameter data in the generated
+ * module.
+ */
+class LinkedParamNode : public Object {
```

Comment:
For my understanding : Why is this moved in this PR ? Can we keep it in a codegen associated class ?

Reply:
Was proposed here:
https://github.com/apache/tvm/pull/8509#discussion_r785371147